howso/utilities/feature_attributes/base.py

```diff
@@ -774,6 +774,14 @@ def _process(self,  # noqa: C901
 
         self.default_time_zone = default_time_zone
 
+        # Store ID feature names, if providded, for processing
+        if isinstance(id_feature_name, str):
+            self.id_feature_names = [id_feature_name]
+        elif isinstance(id_feature_name, Iterable):
+            self.id_feature_names = id_feature_name
+        else:
+            self.id_feature_names = []
```

Comment:
There is a similar check like this further down. May be a case to consolidate them

Reply:
Ah wait nvm, I see that code needs to be after the infer and this is needed before.